packages/application-shell/src/components/navbar/use-navbar-state-manager.ts

```diff
@@ -125,9 +127,14 @@ const useNavbarStateManager = (props: HookProps) => {
     STORAGE_KEYS.IS_FORCED_MENU_OPEN
   );
   const isForcedMenuOpen = isNil(cachedIsForcedMenuOpen)
-    ? null
+    ? isForcedMenuOpenDefaultValue
     : (JSON.parse(cachedIsForcedMenuOpen) as boolean);
 
+  window.localStorage.setItem(
+    STORAGE_KEYS.IS_FORCED_MENU_OPEN,
+    String(isForcedMenuOpen)
+  );
```

Comment:
Should we write it only if it wasn't set before?

```suggestion
  if (!cachedIsForcedMenuOpen) {
    window.localStorage.setItem(
      STORAGE_KEYS.IS_FORCED_MENU_OPEN,
      String(isForcedMenuOpen)
    );
  }
```

Reply:
Yeah I think we need to set it only if it wasn't set, otherwise if the user expands the menu and reloads the page, the value will always be overwritten with `false`.

Reply:
Yep, thank you

Reply:
Sorry, maybe I'm missing something...

We first read the value from the localeStorage.
Then we calculate `isForcedMenuOpen` based on that value.
Afterwards, we store that calculated value in the localStorage, only if it's truthy.

So, the value won't be stored in the the localStorage if we didn't have any value in localStorage before (`cachedIsForcedMenuOpen` would be `null`) or if the value over there was already `false`.

Perhaps what would would like to do is to store the calculated value in the localStorage if it's different to what we didn't have any value over there before.

```
  if (cachedIsForcedMenuOpen === null) {
    window.localStorage.setItem(
      STORAGE_KEYS.IS_FORCED_MENU_OPEN,
      String(isForcedMenuOpen)
    );
  }
```

What do you think?

Reply:
Yes good point, it should be `cachedIsForcedMenuOpen === null` 💯